lib/datasource/local/settings_storage.dart

```diff
@@ -169,22 +177,30 @@ class _SettingsStorage {
   void enableRecoveryMode({required String accountName, String? privateKey}) {
     inRecoveryMode = true;
     _accountName = accountName;
-    this.privateKey = privateKey;
+    _privateKey = privateKey;
   }
 
   void finishRecoveryProcess() => inRecoveryMode = false;
 
   void cancelRecoveryProcess() {
     inRecoveryMode = false;
     _accountName = null;
-    privateKey = null;
+    _privateKey = null;
   }
 
   void savePasscode(String? passcode) => this.passcode = passcode;
 
-  void saveAccount(String accountName, String privateKey) {
+  Future<void> saveAccount(String accountName, String privateKey) async {
     _accountName = accountName;
-    this.privateKey = privateKey;
+    _privateKey = privateKey;
+    // Retrieve private keys list
+    final List<String> pkeys = (await _secureStorage.read(key: _kPrivateKeysList))?.split(',') ?? [];
+    // Add new private key
```

Comment:
same, need to make sure that the pkey is not already in the list - we never want to store duplicated list items

Reply:
Done